### DISCUSSION TITLE
Plan and Execute

### DISCUSSION THREAD

**Discussion by nhorton** [patterns-ai-core/langchainrb#636] [General]
It seems like none of the LLM wrapper projects in Ruby have examples doing Plan and Execute (or similar architectures). Is there one expected for LangchainRb anytime soon? 3rd-party examples would be fine too.

**Comment by andreibondarev** [comment]
@nhorton If you don't mind me asking -- For your use-case, is the LLM interfacing with an end-user/human?

In one of the previous gem versions we had an implementation of a [ReAct](https://arxiv.org/abs/2210.03629) agent: 
* [ReActAgent code](https://github.com/patterns-ai-core/langchainrb/blob/0.10.0/lib/langchain/agent/react_agent.rb)
* [ReActAgent prompt](https://github.com/patterns-ai-core/langchainrb/blob/0.10.0/lib/langchain/agent/react_agent/react_agent_prompt.yaml)

Depending on your use-case, the newer Langchain::Assistant might fit your needs.

**Reply by nhorton** [reply]
Hey,
We have a mix of scenarios with human interaction and not. 

Assistants is definitely the a construct we would use if we switch over to langchainrb, but we need complex planning too. E.g. we use OpenAI assistants already and exceed the capability of of the iterative planning and error correction that ones gets with just straight Function invocation.

**Reply by andreibondarev** [reply]
@nhorton We can put together an Agent based on the [Plan and Execute architecture](https://blog.langchain.dev/planning-agents/#plan-and-execute) (or similar) relatively quickly, this weekend perhaps.

Is this something you would be interested in trying out and providing feedback? I'd love to work closely together on it to make sure it meets all of your needs. If preferred, we can move this conversation to email: andrei@sourcelabs.io
